lib/adapter.js

```diff
@@ -2040,6 +2046,28 @@ function Adapter(options) {
                 return tools.maybeCallbackWithError(callback, tools.ERRORS.ERROR_DB_CLOSED);
             }
 
+            // remove the preserve attributes
+            if (oldObj && options && tools.isObject(options.preserve)) {
+                /**
+                 * Removes properties which are given by preserve
+                 * @param {object} preserve - object which has true entries for all attributes which should be removed from currObj
+                 * @param {object} oldObj - old object
+                 * @param {object} newObj - new object
+                 */
+                const rmProperties = (preserve, oldObj, newObj) => {
```

Comment:
Why not put it one level above and use same for both places?

Reply:
hm very specific method only interesting for this extendObject, but we can also provide it outside of the methods

Reply:
I was justthinking because of the duplicated code
@AlCalzone what do you think?

Reply:
We could also put it into tools.js :-)

Reply:
sounds like a decent utility method for tools.